pkg/controller/pravegacluster/upgrade_test.go

```diff
@@ -221,6 +225,51 @@ var _ = Describe("Pravega Cluster Version Sync", func() {
 					Ω(upgradeCondition.Message).Should(Equal("0"))
 				})
 			})
+			Context("Upgrade Segmentstore to empty version", func() {
+				var (
+					foundPravega *v1beta1.PravegaCluster
+				)
+				BeforeEach(func() {
+					foundPravega = &v1beta1.PravegaCluster{}
+					_ = client.Get(context.TODO(), req.NamespacedName, foundPravega)
+					foundPravega.Status.TargetVersion = ""
+
+					foundPravega.Status.SetPodsReadyConditionTrue()
+					client.Update(context.TODO(), foundPravega)
+					_, _ = r.Reconcile(req)
+					foundPravega = &v1beta1.PravegaCluster{}
+					_ = client.Get(context.TODO(), req.NamespacedName, foundPravega)
+
+				})
+
+				It("should set upgrade condition reason and message to empty", func() {
+					_, upgradeCondition := foundPravega.Status.GetClusterCondition(pravegav1beta1.ClusterConditionUpgrading)
+					Ω(upgradeCondition.Reason).Should(Equal(""))
+					Ω(upgradeCondition.Message).Should(Equal(""))
+				})
+			})
+			Context("checking upgrade completion", func() {
+				var (
+					foundPravega *v1beta1.PravegaCluster
+				)
+				BeforeEach(func() {
+					foundPravega = &v1beta1.PravegaCluster{}
+					_ = client.Get(context.TODO(), req.NamespacedName, foundPravega)
+					foundPravega.Status.TargetVersion = "0.7.0"
+
+					foundPravega.Status.SetPodsReadyConditionTrue()
+					foundPravega.Status.CurrentVersion = "0.7.0"
+					client.Update(context.TODO(), foundPravega)
+					_, _ = r.Reconcile(req)
+					foundPravega = &v1beta1.PravegaCluster{}
+					_ = client.Get(context.TODO(), req.NamespacedName, foundPravega)
+				})
+
+				It("should set the target version to empty", func() {
+
+					Ω(foundPravega.Status.TargetVersion).To(BeEquivalentTo(""))
```

Comment:
Lets set Upgrade Condition to true in `BeforeEach` before calling Update(), and check that Upgrade Condition is also being set to false inside the assertion.

Reply:
done